src/Collections/Sites.php

```diff
@@ -140,6 +140,19 @@ public function filterByTag($tag)
         });
     }
 
+    /**
+     * Filters sites list by upstream
+     *
+     * @param string $upstream An upstream to filter by
+     * @return Sites
+     */
+    public function filterByUpstream($upstream_uuid)
```

Comment:
Small note: This upstream can be identified by its label or its machine_name, too.
https://github.com/pantheon-systems/terminus/blob/master/src/Models/Upstream.php#L35